### DISCUSSION TITLE
reorder operation questions

### DISCUSSION THREAD

**Discussion by terrafrost** [Laravel-Backpack/community-forum#908] [Q&A (Help)] [answered]
Quoting https://backpackforlaravel.com/docs/6.x/crud-operation-reorder ,

> Your model should have the following integer fields, with a default value of 0: `parent_id`, `lft`, `rgt`, `depth`.
>
> Additionally, the `parent_id` field has to be nullable.

Should any of those have indexes?

Also, when I set up a re-orderable CRUD controller `setupListOperation()` doesn't show things in the same order that `setupReorderOperation()` does. Is there a way to make it so that it does?

Finally, is there any guidance on how to show things in-order using the schema that Laravel Backpack uses outside of Laravel Backpack but still in the context of Laravel? I mean, I recognize that Laravel Backpack is using a [nested set hierarchy](https://en.wikipedia.org/wiki/Nested_set_model) but I'd just assume use existing code to show stuff in order (eg. pass a sorted collection to a view or some such) rather than re-invent the wheel and write my own code that does that.

**Comment by karandatwani92** [comment, marked as answer]
Hey @terrafrost 

Just add this clause `CRUD::orderBy('lft');` inside `setupListOperation()`.
